configury forces pkg-config to poke for sdkdir variable which 
pkg-config happily returns but does not prepend sysroot to it
since thats only done for includes and libs. This then gets -I
prepended and added to CFLAGS. Guess what gcc cross gripes
about including host includes in cross builds but does not bail
out the build but it does not escape the wrath of our oe's
QA checks and rightly so. We replace use of sdkdir with 
--cflags-only-I which returns the include paths correctly
pointing inside sysroot and not at the host includes and plus
includedir if you look inside the .pc file is assigned sdkdir
anyway so we do not lose anything

This m4 macro is included in other packages e.g. xf86-video-fbdev
which dont build on hosts which have installation of
xorg-xserver-dev because it gets this include poisoning

Signed-off-by: Khem Raj <raj.khem@gmail.com>

Upstream-status: Pending

Index: xorg-server-1.10.1/xorg-server.m4
===================================================================
--- xorg-server-1.10.1.orig/xorg-server.m4
+++ xorg-server-1.10.1/xorg-server.m4
@@ -31,7 +31,7 @@ dnl
 AC_DEFUN([XORG_DRIVER_CHECK_EXT],[
 	AC_REQUIRE([PKG_PROG_PKG_CONFIG])
 	SAVE_CFLAGS="$CFLAGS"
-	CFLAGS="$CFLAGS -I`$PKG_CONFIG --variable=sdkdir xorg-server`"
+	CFLAGS="$CFLAGS `$PKG_CONFIG --cflags-only-I xorg-server`"
 	AC_COMPILE_IFELSE([AC_LANG_PROGRAM([[
 #include "xorg-server.h"
 #if !defined $1
